discovery_engine_core/web-api/src/models.rs

```diff
@@ -77,7 +77,7 @@ impl Reject for Error {}
     sqlx::FromRow,
 )]
 #[sqlx(transparent)]
-pub struct DocumentId(String);
+pub struct DocumentId(pub String);
```

Comment:
we might not need inner value to be `pub` if we derive `AsRef`

Reply:
see my other comment to sum up:

- `pub` allows creating a malformed `DocumentId` instance (by constructor and by modification)
- combining `AsRef` (or `Deref`) with `impl From<DocumentId> for String` allows read only access and
  destruction without that drawback.